src/timezone_tools/_converter.py

```diff
@@ -278,10 +280,11 @@ def first_day_of_month(
                 timezone to localize from. Use `make_aware` to make a naive
                 datetime timezone-aware.
         """
-        return self.date(datetime).replace(day=1)
+        date = self.midnight(when)
```

Comment:
(nitpick) This value is a datetime, not a date, so it's a bit confusing to call it `date`.

Reply:
Fair!